packages/coingecko/src/index.ts

```diff
@@ -0,0 +1,25 @@
+import { useEffect, useState } from 'react'
+import { useBlockNumber } from '@usedapp/core'
+import { getCoingeckoPrice } from './helpers'
+
+export const useCoingeckoPrice = (base: string, quote = 'usd'): number | undefined => {
+  const [price, setPrice] = useState(undefined)
+  const blockNo = useBlockNumber()
+
+  useEffect(() => {
+    async function getPrice() {
+      try {
+        const tokenPrice = await getCoingeckoPrice(base, quote)
+        setPrice(tokenPrice)
+      } catch (_) {
+        setPrice(undefined)
+      }
+    }
+
+    if (base && quote && blockNo) {
+      getPrice()
+    }
```

Comment:
You can just call `getPrice()` without if since base and quote are required in hook and even if function throws it will just set price to undefined